### PR TITLE
refactor(openapi): add from __future__ import annotations

### LINKED ISSUE #4991: Tech debt: add `from __future__ import annotations` in the Event Handler OpenAPI feature
### Why is this needed?

Reference: https://github.com/aws-powertools/powertools-lambda-python/issues/4607

Add from future import annotations to all files, including examples

### Which area does this relate to?

Static typing

### Suggestion

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by ericbn**
**Issue number:** #4991 

### Changes

Add `from __future__ import annotations` to openapi package

### User experience

Discussed in #4607

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [ ] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by ericbn** [comment]
@leandrodamascena, looks like Pydantic does not work with the new type annotations out-of-the-box with `from __future__ import annotations`, which actually makes sense. I'm getting this error e.g. with Python 3.8:

> TypeError: You have a type annotation 'str | None' which makes use of newer typing features than are supported in your version of Python. To handle this error, you should either remove the use of new syntax or install the `eval_type_backport` package.

**Comment by leandrodamascena** [comment]
> @leandrodamascena, looks like Pydantic does not work with the new type annotations out-of-the-box with `from __future__ import annotations`, which actually makes sense. I'm getting this error e.g. with Python 3.8:
> 

I think it works, but not in version 3.8/3.9. I'm running tests with Python 3.10 and 3.11 and it's working perfectly.

I don't think we have much choice here other than not to change the openapi utility. I wouldn't want to install anything new and/or refactor and risk breaking something.

I don't see Pydantic being able to fix this, as it seems more like a problem with older versions of Python. We can close this PR and keep the issue open as Tech Debt so that it can be fixed one day.

What do you think?

**Comment by ericbn** [comment]
The code in this PR will fail in Python 3.8/3.9. I'm going to fix it.

**Comment by ericbn** [comment]
Added a new commit to fix Python 3.8/3.9

**Comment by leandrodamascena** [comment]
> Added a new commit to fix Python 3.8/3.9

I saw that you reverted some files. Let's wait for CI to complete.
The error in SonarCloud is expected and we don't need to care about this.

File: aws_lambda_powertools/event_handler/openapi/types.py
Comment (leandrodamascena) [inline aws_lambda_powertools/event_handler/openapi/types.py:13]: Smart catch 😃

**Comment by leandrodamascena** [comment]
Hi @ericbn! I see that the CI is green now and I've done an initial review and everything looks good. Tomorrow morning I'll do one last review before approving and merging.

I can't thank you enough for all the effort you've put into these PRs, it's an incredible amount of amazing work. ❤️

**Review by leandrodamascena — APPROVED** [review]
APPROVED!

**Pull request merged**
